modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/AttachmentDTOConverter.java

```diff
@@ -77,6 +80,35 @@ public Attachment toDTO(DTOConverterContext dtoConverterContext)
 				setExpirationDate(cpAttachmentFileEntry::getExpirationDate);
 				setExternalReferenceCode(
 					cpAttachmentFileEntry::getExternalReferenceCode);
+				setFileEntryExternalReferenceCode(
+					() -> {
+						FileEntry fileEntry =
+							cpAttachmentFileEntry.fetchFileEntry();
```

Comment:
is used in two points, please move this on top and leave inside these methods only the check of null and the return